src/test/java/agh/ics/oop/AnimalTest.java

```diff
@@ -0,0 +1,87 @@
+package agh.ics.oop;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static agh.ics.oop.OptionsParser.parse;
+import static org.junit.jupiter.api.Assertions.*;
+
+class AnimalTest {
+    Animal animal;
+
+    @BeforeEach
+    void setUp() {
+        animal = new Animal();
+    }
+
+    @Test
+    void moveTest() {
+        String[] args = {"f", "r", "b", "l", "b", "left", "right", "backward", "forward", "for", "back", "bi", "ri", "test", " ", ""};
+        List<MoveDirection> directions = parse(args);
+        for (MoveDirection direction : directions) {
+            animal.move(direction);
+        }
+        assertAll(
+                () -> assertEquals(MapDirections.NORTH, animal.getDirection()),
+                () -> assertTrue(animal.isAt(new Vector2d(1, 2)))

```

Comment:
tu trochę dużo założeń czynimy, jeśli po (długiej) drodze kod robiłby błędy, ale wykonał ich "parzystą ilość" (czyli zniwelował kolejnymi błędami skutki poprzednich) to test nadal by błędnie przechodził. Trzeba by jednak sprawdzić kroki pośrednie, niekoniecznie robiąc tak długi przypadek testowy.